File: src/main/java/ru/ioleksiv/telegram/bot/core/model/objects/inline/InlineQuery.java
package ru.ioleksiv.telegram.bot.core.model.objects.inline;

import com.fasterxml.jackson.annotation.JsonProperty;
import ru.ioleksiv.telegram.bot.core.model.objects.std.Location;
import ru.ioleksiv.telegram.bot.core.model.objects.std.User;

/**
 * @see <a href="https://core.telegram.org/bots/api#inlinequery>InlineQuery</a>
 */
public class InlineQuery {
    /**
     * id	String	Unique identifier for this query
     */
    @JsonProperty("id")
    private String id = null;
    /**
     * from	User	Sender
     */
    @JsonProperty("from")
    private User from = null;
    /**
     * location	Location	Optional. Sender location, only for bots that request user location
     */
    @JsonProperty("location")
    private Location location = null;
    /**
     * query	String	Text of the query (up to 512 characters)
     */
    @JsonProperty("query")
    private String query = null;
    /**
     * offset	String	Offset of the results to be returned, can be controlled by the bot
     */
    @JsonProperty("offset")
    private String offset = null;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public User getFrom() {
        return from;
    }

    public void setFrom(User from) {
        this.from = from;
    }

    public Location getLocation() {
        return location;
    }

    public void setLocation(Location location) {
        this.location = location;
    }

    public String getQuery() {
        return query;
    }

    public void setQuery(String query) {
        this.query = query;
    }

    public String getOffset() {
        return offset;
    }

    public void setOffset(String offset) {
        this.offset = offset;
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/objects/inline/InlineQuery.kt
package io.github.oleksivio.tl.kbot.server.api.objects.inline

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.model.objects.IUserFrom
import io.github.oleksivio.tl.kbot.server.api.objects.std.Location
import io.github.oleksivio.tl.kbot.server.api.objects.std.User

/**
 *  [InlineQuery](https://core.telegram.org/bots/api/#inlinequery)
 */
data class InlineQuery(
    /**
     * id String Unique identifier for this query
     */
    @JsonProperty("id")
    var id: String? = null,
    /**
     * To setup filter:
     *
     *  UserFilter from from User Sender
     */
    @JsonProperty("from")
    override var from: User?,
    /**
     * To setup filter:
     *
     *  LocationFilter location location Location Optional. Sender location, only for bots that request user location
     */
    @JsonProperty("location")
    var location: Location? = null,
    /**
     * To setup filter:
     *
     *  StringFilter query query String Text of the query
     */
    @JsonProperty("query")
    var query: String? = null,
    /**
     * To setup filter:
     *
     *  StringFilter offset offset String Offset of the results to be returned, can be controlled by the bot
     */
    @JsonProperty("offset")
    var offset: String? = null
) : IUserFrom
